pkgdb/tests/lock-flake-installable.cc

```diff
@@ -0,0 +1,162 @@
+
+  return true;
+}
+
+/**
+ * @brief Test that the flake origin is correctly parsed from the flake
+ */
+bool
+test_flakerefOrigins( const nix::ref<nix::EvalState> & state,
+                      const std::string &              system )
+{
+  auto githubScheme = flox::lockFlakeInstallable( state,
+                                                  system,
+                                                  "github:nixos/nixpkgs/"
+                                                    + nixpkgsRev + "#hello" );
+  auto gitHttpsScheme
+    = flox::lockFlakeInstallable( state, system, nixpkgsRef + "#hello" );
+
+
+  return true;
+}
+
+/**
+ * @brief Test that the flake origin is correctly parsed from the flake
+ */
+bool
+test_locksUrl( const nix::ref<nix::EvalState> & state,
+               const std::string &              system )
+{
+  auto unlockedUrl = "github:nixos/nixpkgs#hello";
+  auto lockedInstallable
+    = flox::lockFlakeInstallable( state, system, unlockedUrl );
+
+
+  EXPECT( nix::parseFlakeRef( lockedInstallable.lockedUrl ).input.isLocked() );
+
+  return true;
+}
+
+bool
+test_explicitOutputs( const nix::ref<nix::EvalState> & state,
+                      const std::string &              system )
+{
+
+  auto defaultOutputs
+    = flox::lockFlakeInstallable( state,
+                                  system,
+                                  "github:nixos/nixpkgs#openssl" );
+
+  // Default outputs of openssl are `bin` and `man`
+  EXPECT( defaultOutputs.outputsToInstall
+          == nix::StringSet( { "bin", "man" } ) );
+
+  auto explicitOutputs
+    = flox::lockFlakeInstallable( state,
+                                  system,
+                                  "github:nixos/nixpkgs#openssl^dev,doc" );
+
+  EXPECT( explicitOutputs.outputsToInstall
+          == nix::StringSet( { "dev", "doc" } ) );
+
+  auto allOutputs
+    = flox::lockFlakeInstallable( state,
+                                  system,
+                                  "github:nixos/nixpkgs#openssl^*" );
+  debugLog( "allOutputs.outputsToInstall: "
+            + nlohmann::json( allOutputs.outputsToInstall ).dump() );
+
+  EXPECT( allOutputs.outputsToInstall
+          == nix::StringSet( { "bin", "dev", "out", "man", "doc" } ) );
+  return true;
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+int
+main( int argc, char * argv[] )
+{
+  int exitCode = EXIT_SUCCESS;
+#define RUN_TEST( ... ) _RUN_TEST( exitCode, __VA_ARGS__ )
+
+  nix::verbosity = nix::lvlDebug;
+  if ( ( 1 < argc ) && ( std::string_view( argv[1] ) == "-v" ) )  // NOLINT
+    {
+      nix::verbosity = nix::lvlDebug;
+    }
+
+  /* Initialize `nix' */
+  flox::NixState nstate;
+  auto           state = nstate.getState();
+
+  std::string system = nix::nativeSystem;
+
+  RUN_TEST( attrpathUsesDefaults, state, system );
+  RUN_TEST( flakerefOrigins, state, system );
+  RUN_TEST( locksUrl, state, system );
+  RUN_TEST( explicitOutputs, state, system );
```

Comment:
blocking: not sure if you were done with tests anyways, but there isn't coverage for
- defaulting to `packages.system.default`
- scraping most of the fields
- a bunch of the error cases, although I'm not sure we need coverage for those at this point

Reply:
i think we have covered all the scraped fields.
I'd like to see how detailed we neeed to catch errors, right now they are sjut thrown forward.
I guess technically all accessors can fail if their type is incorrect e.g. `version = 1` and we might eventually catch all of those as well. Im unsure however how much time we need to put into this right now